### PR TITLE
Feat/cmdk in docs

### PR DESCRIPTION
**Pull request by MildTomato**
## What kind of change does this PR introduce?

- Added new CMDK to docs
  - Conversation style AI. although AI currently doesn't understand last answers, need to update in 
  - Docs search now has updated styling, keyboard navigation and debounce searching
  - Lots of static links added
  - Can toggle theme light/dark
  
Screenshot 2023-03-30 at 2 42 44 PM

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/docs/pages/_app.tsx [hunk 1]
```diff
@@ -68,28 +70,44 @@ function MyApp({ Component, pageProps }: AppPropsWithLayout) {
         <SessionContextProvider supabaseClient={supabase}>
           <AuthProvider>
             <ThemeProvider>
-              <SearchProvider>
-                {/* <CommandMenuProvider site="docs"> */}
+              <CommandMenuProvider
+                site="docs"
+                MarkdownHandler={({ children }: { children: string }) => (
+                  <ReactMarkdown
```
Comment (MildTomato) [inline apps/docs/pages/_app.tsx:76]: will move somewhere else so _app isn't clogged up

**Review by MildTomato — COMMENTED** [review]
comments

File: apps/docs/components/Navigation/NavBar.tsx [hunk 2]
```diff
@@ -3,12 +3,12 @@ import Image from 'next/image'
...
 import { useRouter } from 'next/router'
 import { FC, useEffect, useState } from 'react'
-import { IconCommand, IconMenu, IconMoon, IconSearch, IconSun, Listbox } from 'ui'
+import { IconCommand, IconMenu, IconMoon, IconSearch, IconSun, Listbox, SearchButton } from 'ui'
 import { REFERENCES } from './Navigation.constants'
 
 import { getPageType } from '~/lib/helpers'
 
-import SearchButton from '../Search/SearchButton'
+// import SearchButton from '../Search/SearchButton'
```
Comment (joshenlim) [inline apps/docs/components/Navigation/NavBar.tsx:11]: remove commented out line, same for TopNavBar + TopNavBarRef

Comment (MildTomato) [reply apps/docs/components/Navigation/NavBar.tsx]: done

File: apps/docs/pages/_app.tsx [hunk 3]
```diff
@@ -68,28 +70,32 @@ function MyApp({ Component, pageProps }: AppPropsWithLayout) {
...
                   <Component {...pageProps} />
                 </SiteLayout>
-                {/* </CommandMenuProvider> */}
-              </SearchProvider>
+              </CommandMenuProvider>
             </ThemeProvider>
           </AuthProvider>
         </SessionContextProvider>
       ) : (
-        <AuthProvider>
```
Comment (joshenlim) [inline apps/docs/pages/_app.tsx:82]: @alaister do we need the AuthProvider here? i remember in your previous PR we declared it here

File: apps/docs/pages/_app.tsx [hunk 4]
```diff
@@ -61,36 +63,33 @@ function MyApp({ Component, pageProps }: AppPropsWithLayout) {
 
   const SITE_TITLE = 'Supabase Documentation'
 
+  const AuthContainer = (props) => {
+    return IS_PLATFORM || LOCAL_SUPABASE ? (
+      <SessionContextProvider supabaseClient={supabase}>
+        <AuthProvider>{props.children}</AuthProvider>
```
[the remark this answers is not in the document]
Comment (alaister) [reply apps/docs/pages/_app.tsx:69]: Yeah, nice catch @joshenlim. We can think of `<AuthProvider />` as the dashboard auth state, so anywhere we want that we should have it. Do we have self-hosting for the docs?

**Review by saltcod — APPROVED** [review]

**Pull request merged**